Sources/SwiftProtobuf/Message+FieldMask.swift

```diff
@@ -0,0 +1,105 @@
+// Sources/SwiftProtobuf/Message+FieldMask.swift - Message field mask extensions
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Extend the Message types with FieldMask utilities.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+extension Message {
+
+  /// Checks whether the given path is valid for Message type.
+  ///
+  /// - Parameter path: Path to be checked
+  /// - Returns: Boolean determines path is valid.
+  public static func isPathValid(
+    _ path: String
+  ) -> Bool {
+    var message = Self()
+    return message.hasPath(path: path)
+  }
+
+  internal mutating func isPathValid(
+    _ path: String
+  ) -> Bool {
+    hasPath(path: path)
+  }
+}
+
+/// Defines available options for merging two messages.
+public enum MergeOption: Equatable {
```

Comment:
I think the better choice might be to model this as a struct, that way you don't end up having a parameter else where that is an array of these. It also brings things in line with how most of the other *Options types in this library have worked. For the one option you have right now, it could then become a boolean to control replace vs. append for repeated and map fields.